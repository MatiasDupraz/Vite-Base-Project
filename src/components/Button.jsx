import React from "react";
import classNames from "classnames";

const Button = ({ button, toggleSuccess }) => {
  let successState = null;
  if (button) {
    if (button.success) {
      successState = true;
    } else if (button.success === false) {
      successState = false;
    }
  }

  const label = successState ? "Success" : "Error";
  const buttonClassName = classNames(
    "w-20 h-10 border-white rounded-md hover:text-white ease-linear duration-300 transition-all",
    successState ? "bg-green-900" : "bg-red-900"
  );
  const simpleButtonClassName = classNames(
    "w-20 h-10 border-white rounded-md hover:text-white ease-linear duration-300 transition-all bg-white hover:bg-neutral-900"
  );
  return successState || successState === false ? (
    <button className={buttonClassName} onClick={toggleSuccess}>
      {label}
    </button>
  ) : (
    <button className={simpleButtonClassName}>{label}</button>
  );
};

export default Button;
