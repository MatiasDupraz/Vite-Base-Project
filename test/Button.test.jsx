import React from "react";
import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import Button from "../src/components/Button";

// In this test example we are calling a button with and without props
// We are also calling a hook and passing down the state and the toggle function, this is because useState cannot be called directly in the tests

const ButtonWrapper = ({ initialSuccess = true }) => {
  const [button, setButton] = React.useState({ success: initialSuccess });
  const toggleSuccess = () => {
    setButton((prevButton) => ({
      ...prevButton,
      success: !prevButton.success,
    }));
  };

  return <Button toggleSuccess={toggleSuccess} button={button} />;
};

describe("<Button />", () => {
  it("should match snapshot without props", () => {
    render(<Button />);
    expect(screen.getByRole("button")).toMatchSnapshot();
  });

  it("should match snapshot with primary prop", () => {
    render(<ButtonWrapper />);
    expect(screen.getByRole("button")).toMatchSnapshot();
  });
});
