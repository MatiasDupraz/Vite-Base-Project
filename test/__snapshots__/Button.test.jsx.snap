// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Button /> > should match snapshot with primary prop 1`] = `
<button
  class="w-20 h-10 border-white rounded-md hover:text-white ease-linear duration-300 transition-all bg-green-900"
>
  Success
</button>
`;

exports[`<Button /> > should match snapshot without props 1`] = `
<button
  class="w-20 h-10 border-white rounded-md hover:text-white ease-linear duration-300 transition-all bg-white hover:bg-neutral-900"
>
  Error
</button>
`;
